pkg/cluster/internal/providers/podman/network.go

```diff
@@ -48,6 +49,13 @@ func ensureNetwork(name string) error {
 		return nil
 	}
 
+	// in case kind is running in linux and IPv6 is disabled in linux kernel, then create the podman network without Ipv6
+	// This is required here because i came across that even though ipv6 is disabled in kernel, podman doesn't stop you from creating a network with Ipv6 enabled,
+	// which later will result in crashing when it'll try to create the cluster.
+	if isIPv6DisabledInLinuxKernel() {
+		return createNetwork(name, "")
```

Comment:
we rely on the errors returned, see Lines 69-73
```
	if isUnknownIPv6FlagError(err) ||
		isIPv6DisabledError(err) {
```

if podman creates a network without ipv6 when we ask it to create it with ipv6 it looks a podman bug

Reply:
@aojea 
podman is  creating the network with ipv6, the problem is that he does that even if the linux kernel ipv6 is disabled.

I Have ipv6 disabled in linux kernel,
```shell
[zgrinber@zgrinber kind]$ cat /sys/module/ipv6/parameters/disable
1

```

But trying creating a cluster with kind and podman failed, as you can see in the following printscreen:

![image](https://github.com/kubernetes-sigs/kind/assets/75700623/9e8b238d-af39-4a44-b462-4be613b583c1)

and with my fix it succeeded:

![image](https://github.com/kubernetes-sigs/kind/assets/75700623/8933f461-83a3-4f1e-9a23-e20067d71a45)

![image](https://github.com/kubernetes-sigs/kind/assets/75700623/1230e24a-4a1c-46f3-b495-30d8a1628237)

Reply:
@aojea
The Real problem in the above scenario, that it's not even  reaching code in lines 69-73 that you mentioned, as it's creating successfully the network:
```go
	subnet := generateULASubnetFromName(name, 0)
	err := createNetwork(name, subnet)
// this one causes to skip the treatment at Lines 69-73
	if err == nil {
		// Success!
		return nil
	}
```

Reply:
@aojea I Believe that there are scenarios/use cases where podman returns errors and others where it doesn't return error...., and the second ones are captured by my treatment, whereas the first ones are captured by the treatment atlines 69-73

Reply:
> as it's creating successfully the network:

why podman creates the network if ipv6 is disabled? that sounds like a bug

Reply:
I Really Don't know, and it's not consistent , in some cases it returns error and in others it creates the network despite ipv6 is disabled....
My best smart guess is that sysctl net.ipv6.conf.all.disable_ipv6 property doesn't exists on my machine/system:
```shell
[zgrinber@zgrinber kind]$ sysctl net.ipv6.conf.all.disable_ipv6
sysctl: cannot stat /proc/sys/net/ipv6/conf/all/disable_ipv6: No such file or directory
```
instead of returning 0 or 1.
Maybe this cause podman to create the network possibly it reads that value

Reply:
I think the point @aojea is making is: this is a bug in podman and it should be fixed in podman. Kind should not have special handling to try to work around this issue.